app/models/bundle_instance.rb

```diff
@@ -0,0 +1,12 @@
+class BundleInstance < ApplicationRecord
+  belongs_to :bundle
+  belongs_to :site
+  has_many :application_instances
+  has_many :applications, through: :bundle
+
+  def create_application_instances
+    bundle&.applications&.map do |app|
```

Comment:
Just when you don't need to return an array, it is generally preferable to do `.each` so there isn't as much confusion. Maybe @jbasdf has an opinion?